activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -91,7 +91,7 @@ def self.build_read_query_regexp(*parts) # :nodoc:
       end
 
       def self.find_cmd_and_exec(commands, *args) # :doc:
-        commands = Array(commands)
+        commands = Array(commands).compact
```

Comment:
Rather than needing to the first value is `nil`, there should be a default for each db set. This shouldn't have to call compact.